haris2055/SOEN390
# [EPIC 10] Admin features

This pull request includes the implementation of user stories featured in #80 [EPIC 10] Admin:

#147 [US-52] Handle User Harassment Reports
To test this feature:
- Reporting a user
  - go to login page
  - sign in to a valid account
  - navigate to another user's profile page
  - press the Report button, enter a message into the reporting form and press submit

- Managing user reports
  - ensure user reports have been submitted against one or more users
  - go to login page
  - sign in to a valid account that has Admin privileges
  - in the header's Admin tab, navigate to Moderate Users
  - interact with the buttons on each reported User (view the user profile, dismiss the report, etc.) 

#148 [US-53] Admin Can Ban a User
- ensure user reports have been submitted against one or more users
- go to login page
- sign in to a valid account that has Admin privileges
- in the header's Admin tab, navigate to Moderate Users
- press the 'Ban User' button next to the user you are trying to ban
- verify that the banned user's posts do not appear in the newsfeed
- verify that the banned user's jobs do not appear in the jobs feed
- verify that the banned user's jobs do not appear in search results
- verify that the banned user does not appear in search results

#149 [US-54] Admin Can Delete Posts
- Reporting a post
  - go to login page
  - sign in to a valid account
  - ensure you are a connected to another user that has created posts
  - from the newsfeed, press the Report Post button from the dropdown menu next to the post
  - enter a message into the reporting form, and press submit

- Managing post reports
  - ensure post reports have been submitted against one or more posts
  - go to login page
  - sign in to a valid account that has Admin privileges
  - in the header's Admin tab, navigate to Moderate Posts
  - interact with the buttons next to each Post Report (dismiss the report, delete the post, etc.) 

#150 [US-55] Admin Can Delete Job Posts
- Reporting a job post
  - go to login page
  - sign in to a valid account
  - ensure you are a connected to another user that has created job posts
  - from the newsfeed, press the Report Post button from the dropdown menu next to the job post
  - enter a message into the reporting form, and press submit

- Managing job post reports
  - ensure job post reports have been submitted against one or more job posts
  - go to login page
  - sign in to a valid account that has Admin privileges
  - in the header's Admin tab, navigate to Moderate Jobs
  - interact with the buttons next to each Job Post Report (dismiss the report, delete the job post, etc.) 

If merged, this pull request closes #80, closes #147, closes #148, closes #149, closes #150.
